### PR TITLE
Replace homepage search w/ full text

### LINKED ISSUE #514: Replace topic search with full text search on home page.
Per the [latest designs](https://gsa.invisionapp.com/share/QUDR1NBCK#/screens) and as part of #448, we'd like to replace the Topic-based search on the homepage with a full text search. We'll want to tweak the copy a bit to match that design.

Don't worry about the "View all topics" link.

### PR DESCRIPTION
**Pull request by cmc333333**
This resolves #514 by swapping out the search component from the topic auto-completer to a full-text search. It does *not* address the multiple search box problem (#513).

@jeremiak / @rememberlenny do you have any suggestions for how/if to test this? Is there a cleaner way to split up the components?

@micahtaylor I know the designs are still in flux, but is this a step in the right direction?

## Looks like
screen shot 2017-10-04 at 3 44 21 pm

### CODE REVIEW

File: ui/components/homepage/search.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { withRouter } from 'next/router';
+import React from 'react';
+
+import { Search } from '../search/search';
+
+export class HomepageSearch extends Search {
```
[the remark this answers is not in the document]
Comment (cmc333333) [reply ui/components/homepage/search.js]: That makes sense, I'll take a stab at it.

**Comment by cmc333333** [comment]
@jeremiak when you have a moment, I reworked this. Apologies for the rebase; reusing the component is pretty different approach from before.

**Comment by cmc333333** [comment]
Thanks @rememberlenny! Sorry for the confusion.

**Comment by cmc333333** [comment]
I'm going to rebase and merge. Thanks @jeremiak and @rememberlenny !

**Comment by jeremiak** [comment]
Hey @cmc333333 sorry about not giving a merge. I had one queued up but I forgot that you have to confirm the review for Github to post it. In the end, it looks good to me! Though I'm curious to know what you thought of this approach? Does it feel more simple about than an inheritance chain?

**Comment by cmc333333** [comment]
@jeremiak absolutely! I think this is more evidence that we need to stop using the basscss classes in our JSX.

**Pull request merged**